src/main/java/javaslang/collection/Stream.java

```diff
@@ -1541,11 +1541,6 @@ private Nil() {
         }
 
         @Override
-        public boolean containsSlice(Iterable<? extends T> that) {
```

Comment:
For user-friendly error messages I check for null when args are other than simple values.

Here I would insert `Objects.requireNonNull(that, "that is null");`

and also declare a javadoc param `@throws NullPointerException if that is null`.